internal/nexus_operations.go

```diff
@@ -0,0 +1,23 @@
+package internal
+
+import (
+	"context"
+
+	"go.temporal.io/sdk/log"
+)
+
+// NexusOperationContext is an internal only struct that holds fields used by the temporalnexus functions.
+type NexusOperationContext struct {
+	Client    Client
+	TaskQueue string
+	Log       log.Logger
+}
+
+// nexusOperationContextKey is a key for associating a [NexusOperationContext] with a [context.Context].
+var nexusOperationContextKey = struct{}{}
+
+// NexusOperationContextFromGoContext gets the [NexusOperationContext] associated with the given [context.Context].
+func NexusOperationContextFromGoContext(ctx context.Context) (nctx *NexusOperationContext, ok bool) {
```

Comment:
Would call it `GetNexusOperationContext` similar to how the `activity` package has `GetX` calls that accept the context. Also, I would suggest not returning a boolean and just letting the result be nil if not in an operation context. Also, I would suggest making a new struct with fields from what's in context each time to discourage mutation.

Alternatively, consider `GetNexusOperationTemporalClient`, `GetNexusOperationInfo` (with just `TaskQueue` in it for now), and `GetNexusOperationLogger` to more consistently match what `activity` package does.

Reply:
This is all internal so I wouldn't worry about it too much at the moment, we can expose some of this later as we see fit.

Reply:
Oh, my mistake. I think we will want to provide it to users one day, but if not exposed today, no prob